trieste/models/keras/utils.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import tensorflow as tf
+
+from ...data import Dataset
+
+
+def get_tensor_spec_from_data(dataset: Dataset) -> tuple[tf.TensorSpec, tf.TensorSpec]:
+    """
+    Extract tensor specifications for inputs and outputs of neural network models, based on the
+    dataset. This utility faciliates constructing neural networks.
+
+    :param dataset: A dataset with ``query_points`` and ``observations`` tensors.
+    :return: Tensor specification objects for the ``query_points`` and ``observations`` tensors.
+    :raise ValueError: If the dataset is not an instance of :class:`~trieste.data.Dataset`.
+    """
+    if not isinstance(dataset, Dataset):
+        raise ValueError(
+            f"This function works only on trieste.data.Dataset objects, however got"
+            f"{type(dataset)} which is incompatible."
+        )
+    input_tensor_spec = tf.TensorSpec(
+        shape=(dataset.query_points.shape[-1],),
```

Comment:
why the last dimension only?

Reply:
size of the dataset is not needed for building neural nets, only dimensionality info is needed - I assume here we will only deal with usual matrices, thats why only the last dimension - I can make it more general

Reply:
done, but I'm not sure `Dataset` would currently support that, I think it has an incompatible check at https://github.com/secondmind-labs/trieste/blob/6b2bb0e73649debaac81157f0f9fdb8d3fdfef5b/trieste/data.py#L51